tools/clang/unittests/HLSL/CompilerTest.cpp

```diff
@@ -2054,6 +2056,107 @@ TEST_F(CompilerTest, CompileThenTestPdbUtilsRelativePath) {
   VERIFY_SUCCEEDED(pPdbUtils->Load(pPdb));
 }
 
+// We had a bug where the block addr blocks were not included in the
+// NumBlocks member in the super block. This test makes sure that number
+// is actually correct.
+TEST_F(CompilerTest, CompileThenTestPdbIntegrity) {
+  // Normal small shader
+  std::string source_0 = R"x(
+      [RootSignature("CBV(b1)")]
+      float4 main() : SV_Target {
+        return float4(1,0,0,1);
+      }
+  )x";
+
+  // This is going to compile to a really big shader (and PDB)
+  std::string source_1 = R"x(
+      Texture1D<float> t0 : register(t0);
+      [RootSignature("CBV(b1),DescriptorTable(SRV(t0))")]
+      float main() : SV_Target {
+        float ret = 0;
+        [unroll]
+        for (int i = 0; i < 256; i++) {
+          ret += t0.Load(i);
+        }
+        return ret;
+      }
+  )x";
+
+  CComPtr<IDxcCompiler3> pCompiler;
+  VERIFY_SUCCEEDED(m_dllSupport.CreateInstance(CLSID_DxcCompiler, &pCompiler));
+
+  struct Profile {
+    llvm::StringRef src;
+    std::vector<const WCHAR *> args;
+  };
+
+  Profile profiles[] = {
+      {source_0,
+       {
+           L"/Tps_6_0",
+           L"/Zi",
+       }},
+      {source_0,
+       {
+           L"/Tps_6_0",
+           L"/Zs",
+       }},
+      {source_1, {L"/Tps_6_0", L"/Zi", L"Od"}},
+      {source_1, {L"/Tps_6_0", L"/Zs", L"Od"}},
```

Comment:
I really can't fathom why clang-format does these sort of things. Why are the longer lines kept on a single line? 🤦